File: test-files/javascript/good-example.js
'use strict';

const Calculator = {
    result: 0,
    
    add: function(a, b) {
        if (typeof a !== 'number' || typeof b !== 'number') {
            throw new TypeError('Arguments must be numbers');
        }
        this.result = a + b;
        return this.result;
    },
    
    divide: function(a, b) {
        if (b === 0) {
            throw new Error('Division by zero');
        }
        this.result = a / b;
        return this.result;
    }
};

module.exports = Calculator;

File: test-files/kotlin/good-example.kt
package com.example.calculator

class Calculator {
    private var result: Double = 0.0

    fun add(a: Double, b: Double): Double {
        result = a + b
        return result
    }

    fun divide(a: Double, b: Double): Double {
        require(b != 0.0) { "Division by zero not allowed" }
        result = a / b
        return result
    }

    fun getResult(): Double = result
}

fun main() {
    val calculator = Calculator()
    val sum = calculator.add(5.0, 3.0)
    println("Result: $sum")
}